lib/helper/Appium.js

```diff
@@ -0,0 +1,839 @@
+ --   `appPackage` - Java package of the Android app you want to run
+ --   `appActivity` - Activity name for the Android activity you want to launch from your package.
+ --   `deviceName`: The kind of mobile device or emulator to use
+ --   `platformVersion`: Mobile OS version
+ --   `app` - The absolute local path or remote http URL to an .ipa or .apk file, or a .zip containing one of these. Appium will attempt to install this app binary on the appropriate device first.
+ --   `browserName`: Name of mobile web browser to automate. Should be an empty string if automating an app instead.
+
+ Example:
+
+ ```js
+ {
+   helpers: {
+       WebDriverIO: {
+           desiredCapabilities: {
+               platformName: "Android",
+               appPackage: "com.example.android.myApp",
+               appActivity: "MainActivity",
+               deviceName: "OnePlus3",
+               platformVersion: "6.0.1"
+           },
+           port: 4723,
+           restart: false
+       }
+    }
+ }
+ ```
+ Additional configuration params can be used from <https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md>
+
+
+ ## Access From Helpers
+
+ Receive a Appium client from a custom helper by accessing `browser` property:
+
+ ```js
+ this.helpers['Appium'].browser
+ ```
+ */
+class Appium extends WebdriverIO {
+
+  /**
+   * Appium Special Methods for Mobile only
+   */
+
+  constructor(config) {
+    super(config);
+    webdriverio = requireg('webdriverio');
+
+    // set defaults
+    this.options = {
+      waitForTimeout: 1000, // ms
+      desiredCapabilities: {},
+      restart: true,
+      manualStart: false,
+      timeouts: {
+        script: 1000 // ms
+      }
+    };
+
+    // override defaults with config
+    Object.assign(this.options, config);
+
+    this.options.baseUrl = this.options.url || this.options.baseUrl;
+    this.options.desiredCapabilities.browserName = this.options.browser || this.options.desiredCapabilities.browserName;
+    browserCap = this.options.desiredCapabilities.browserName;
+    this.options.waitForTimeout /= 1000; // convert to seconds
+
+    if (!this.options.desiredCapabilities.platformName) {
+      if (!this.options.url || !this.options.browser) {
+        throw new Error(`
+        WebDriverIO requires at least these parameters
+        Check your codeceptjs config file to ensure these are set properly
+          {
+            "helpers": {
+              "WebDriverIO": {
+                "url": "YOUR_HOST"
+                "browser": "YOUR_PREFERED_TESTING_BROWSER"
+              }
+            }
+          }
+      `);
+      }
+    } else {
+      if ((this.options.desiredCapabilities.platformName == "iOS") && ((!this.options.desiredCapabilities.app || !this.options.desiredCapabilities.deviceName || !this.options.desiredCapabilities.platformVersion) && (!this.options.desiredCapabilities.browserName) && (!this.options.desiredCapabilities.bundleId))) {
```

Comment:
can this be rewritten without this huge if/else/elsif? I think this can be moved to specific functions.